src/mlpack/methods/reinforcement_learning/environment/frozen_lake.hpp

```diff
@@ -0,0 +1,452 @@
+    //    the we can never generate a possible board.
+    while (!valid)
+    {
+      // Randomly choose tiles in the board (discrete random distribution).
+      Board = genBoardHelper(nRows, nCols, platformRate);
+      valid = dfsHelper(Board, nRows, nCols);
+    }
+    return Board;
+  }
+
+  /**
+   * Perform depth-firsth search to see if the environment board has 
+   * a solution or not.
+   * 
+   * @param candidateBoard: the environment board description.
+   * @param nRows: the height of the environment board.
+   * @param nCols: the width of the environment board.
+   * @return true if there is a solution, false otherwise. 
+   */
+  static bool dfsHelper(std::vector<std::vector<char>> candidateBoard, size_t nRows, size_t nCols)
+  {
+    bool visited[nRows][nCols] = {{false}};
+    std::vector<std::array<size_t, 2>> path;
+    path.push_back({0, 0});
+    while (!path.empty())
+    {
+      auto node = path.back();
+      path.pop_back();
+
+      visited[node[0]][node[1]] = true;
+      int directions[4][2] = {{1, 0}, {0, 1}, {-1, 0}, {0, -1}};
+      for (auto direction : directions)
+      {
+        size_t r_new = direction[0] + node[0];
+        size_t c_new = direction[1] + node[1];
+        if (!visited[r_new][c_new])
+        {
+          if (r_new >= nRows || c_new >= nCols)
+              continue;
+          if (candidateBoard[r_new][c_new] == 'G')
+              return true;
+          if (candidateBoard[r_new][c_new] != 'H')
+              path.push_back({r_new, c_new});
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Perform random distribution of tile in the environment board. There are 4 type of tile:
+   * 'S' is the Start Tile, the starting position of the agent.
+   * 'G' is the Goal Tile, the agent can walk into a goal tile to win.
+   * 'F' is a Frozen Tile, the agent can walk on it.
+   * 'H' is a Hole Tile, the agent can fall into it and the game ends.
+   * 
+   * @param m the height of the environment board.
+   * @param n the width of the environment board.
+   * @return a 2d array that describes the environment board. 
+   */
+  std::vector<std::vector<char>> genBoardHelper(size_t nRows, size_t nCols, double platformRate)
+  {
+    std::vector<char> board[nRows];
+    for (size_t i = 0; i < nRows; i++)
+    {
+      for (size_t j = 0; j < nCols; j++)
+      {
+        auto r = arma::randu();
+        if (r < 1 - platformRate)
+          board[i].push_back('H');
+        else
+          board[i].push_back('F');
+      }
+    }
+    board[0][0] = 'S';
+    board[nRows - 1][nCols - 1] = 'G';
+
+    std::vector<std::vector<char>> returnedBoard(board, board + nRows);
+    return returnedBoard;
+  }
+
+  /**
+   * Utilities function to print board description. 
+   * 
+   * @param board the environment board's description.
+   * @param nRows the height of the environment board.
+   * @param nCols the width of the environment board.
```

Comment:
```suggestion
   * @param board Environment board's description.
   * @param nRows Height of the environment board.
   * @param nCols Width of the environment board.
```

The idea is to start with a capital letter right after the parameter.